pisa/core/distribution_maker.py

```diff
@@ -87,17 +99,75 @@ def __init__(self, pipelines, label=None):
             if not isinstance(pipeline, Pipeline):
                 pipeline = Pipeline(pipeline)
             self._pipelines.append(pipeline)
+
+        data_run_livetime = None
+        if set_livetime_from_data:
+            #
+            # Get livetime metadata if defined in any stage in any pipeline
+            #
+            for pipeline_idx, pipeline in enumerate(self):
+                for stage_idx, stage in enumerate(pipeline):
+                    if not (
+                        hasattr(stage, "metadata")
+                        and isinstance(stage.metadata, Mapping)
+                        and "livetime" in stage.metadata
+                    ):
+                        continue
+
+                    if data_run_livetime is None:
+                        data_run_livetime = stage.metadata["livetime"]
+
+                    if stage.metadata["livetime"] != data_run_livetime:
+                        raise ValueError(
+                            "Pipeline index {}, stage index {} has data"
+                            " livetime = {}, in disagreement with"
+                            " previously-found livetime = {}".format(
+                                pipeline_idx,
+                                stage_idx,
+                                stage.metadata["livetime"],
+                                data_run_livetime,
+                            )
+                        )
+            #
+            # Set param `params.livetime` for any pipelines that have it
+            #
+            if data_run_livetime is not None:
+               # data_run_livetime *= ureg.sec
+
+                for pipeline_idx, pipeline in enumerate(self):
+
+                    if "livetime" not in pipeline.params.names:
+                        continue
+
+                    pipeline.params.livetime.is_fixed = True
+
+                    if pipeline.params.livetime != data_run_livetime:
+
+                        logging.warning(
+                            "Pipeline index %d has params.livetime = %f, in"
+                            " disagreement with data livetime = %f defined by"
+                            " data. The pipeline's livetime param will be"
+                            " reset to the latter value and set to be fixed"
+                            " (if it is not alredy).",
+                            pipeline_idx,
+                            float(pipeline.params.livetime.value),
```

Comment:
... and then don't convert to float. It is always useful to display units with the number in case they are disparate.

You could, e.g., have specified `1 common_year` for one and `1 year` for the other; easy mistake to make, but `float()` of each will yield the same value in both cases, rendering the error message unhelpful.

You are right to use `.value` attr, though, that was a bug on my behalf.

Reply:
I used the float conversion because otherwise the warning message wouldn't run properly. Will remove now